flang/lib/Lower/Bridge.cpp

```diff
@@ -1269,7 +1265,7 @@ class FirConverter : public Fortran::lower::AbstractConverter {
           auto &sym = *std::get<Fortran::parser::Name>(assoc.t).symbol;
           const auto &selector =
               *sym.get<Fortran::semantics::AssocEntityDetails>().expr();
-          if (Fortran::evaluate::IsVariable(selector) &&
+          if (Fortran::evaluate::IsVariable(selector) && selector.Rank() &&
```

Comment:
Didn't `!UnwrapWholeSymbolDataRef` already implied `selector.Rank()` ? I am not sure here, I guess complex part reference are scalars where `!UnwrapWholeSymbolDataRef` would return true, but I am wondering what motivated you to add this.

Reply:
The `selector.Rank()` condition enables the use of selector variables that are array elements, such as `x => a(3)` in the following test.  That case works, but needed to be activated.  The `!UnwrapWholeSymbolDataRef` condition enables the use of selector variables that are whole arrays, such as `y => a`.  I'll change the TODO message to refer to array sections.

```
    program p
      integer :: a(5) = 1
      associate (x => a(3), y => a)
        x = x + 5
        y = y + 1
      end associate
      print*, a
    end
```

Result before the change - TODO message.
Result after the change  - 2 2 7 2 2

Reply:
That makes sense, thanks for explaining.